src/audio/pipeline/pipeline-params.c

```diff
@@ -12,7 +12,7 @@
 #include <sof/lib/mm_heap.h>
 #include <sof/compiler_attributes.h>
 #include <sof/list.h>
-#include <sof/spinlock.h>
+#include <zephyr/spinlock.h>
 #include <ipc/stream.h>
```

Comment:
This looks a little bit odd. Does the user of a spinlock needs to know that we are using Zephyr or XTOS? 

My feeling here is that this needs to be agnostic.

The user will just include `<sof/spinlock.h` and will get the spinlock for the specific OS on which SOF is being compiled.

Reply:
@dbaluta I think what this is trying to achieve is to make "native Zephyr" and to increase separation between Zephyr and XTOS builds. That's why the XTOS-wrapper spinlock.h header is now also moved to the xtos-wrapper/include/zephyr/ directory and it now only contains an XTOS Zephyr wrapper. When building with Zephyr that header isn't needed any more and the Zephyr header is included directly

Reply:
This is still WIP, I'm trying to do this in a clean way AND to avoid contamination between headers. More updates later.